### ISSUE TITLE
Port 80 being used by ingress network even if no services are running

### ISSUE THREAD

**Issue by CristianHenzel** [moby/moby#32557]
**Description**

After creating a swarm, on all swarm nodes the port 80 is being taken up by the ingress network and can not be used for anything else, even if no services are running in the swarm.

Here is the relevant output from "iptables -L":

```
[...]
Chain DOCKER-INGRESS (1 references)
target     prot opt source               destination
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:http
ACCEPT     tcp  --  anywhere             anywhere             state RELATED,ESTABLISHED tcp spt:http
[...]
```

Is there any way to disable the port being used up on the whole swarm through some central command?

**Output of `docker version`:**

```
Client:
 Version:      17.03.0-ce
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   3a232c8
 Built:        Tue Feb 28 08:01:32 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.0-ce
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   3a232c8
 Built:        Tue Feb 28 08:01:32 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by aboch** [comment]
Any interesting daemon log ?
I am assuming at some point there was a service publishing port 80, and the cleanup did not succeed when the service was removed.

**Comment by CristianHenzel** [comment]
I assumed that the port was created when the swarm was created, that's why I didn't really look through the logs.
If that is not the case, then there is indeed a high probability that the port remained from a previous service. In that case, is there an easy way to remove it, or do I have to do that through iptables myself?

**Comment by aboch** [comment]
No, at swarm creation and routing mesh deployment no port gets reserved.
For the cleanup, if `netstat -tpln` does not show any socket open for that port, you could try a more graceful cleanup by creating a service which publishes port 80 and then remove it (make sure to add as many replicas as many nodes you have in your swarm).

**Comment by CristianHenzel** [comment]
I tried your suggestion, started up a service with ```--mode global``` to ensure that it is replicated through the whole swarm, confirmed that port 80 is accessible on the nodes and then removed the service, but unfortunately the port rule remained in iptables.
Afterwards, I deleted the iptables rules manually and then recreated the service, and now the iptables rules do not get created anymore.
Is there any way I can debug why this is happening?

**Comment by aboch** [comment]
hmm, how are you creating the service, can you post the command line ?
When the service is running, what is the output of `sudo netstat -tpln | grep dockerd` ?
Any warn/error in the docker daemon log ?

**Comment by CristianHenzel** [comment]
I started the service with:
```docker service create --publish "80:80" --mode global neoof86/hello-world-nginx```

When the service is running, I don't see the port listed in ```netstat -tlpn```, I only see **2377** and **7946** there.
Also, when I start the service, there is nothing in the docker daemon log.
When I stop it, there are some warnings and errors, but I'm not sure how related they are to the problem at hand (msg="task unavailable"), if you want though, I could paste them so you can have a look.

**Comment by aboch** [comment]
> When the service is running, I don't see the port listed in netstat -tlpn, I only see 2377 and 7946 there.

Port 80 is supposed to be there, although not strictly necessary for the request to be sent to the task.

Yes, please attach the daemon logs. There should be some warn/error reported about this.

Can you confirm that when the service is running you can reach your service curling to any of the Host IP addresses:80 ?

**Comment by CristianHenzel** [comment]
I pasted the log [here](https://pastebin.com/KQZUCjWj) (if you want me to paste it somewhere else, please let me know).
I started the service at 19:24:15 and stopped it at 19:26:55.

Also, since I manually deleted the entries from iptables, I can not access anything on port 80 on the machine (no matter if I publish it as mode ingress or mode host), I just get timeouts.

**Comment by aboch** [comment]
That is weird.
Could you please rerun the same service creation, connectivity test, then service deletion after restarting the daemon with debug level logs enabled (`dockerd -D`) ?
And post again the daemon logs somewhere.

Thanks.

**Comment by CristianHenzel** [comment]
After restarting docker in debug mode, as you suggested, I have noticed 2 things:
1. The iptables rule for port 80 was recreated
2. Docker tried communicating with a node, which I have removed a couple of days ago from the swarm

I have then connected to the removed node and noticed it still had some services running, one of which was publishing port 80. So I guess this explains the problem.
My mistake, I think, was to remove the node from the swarm using ```docker node rm```, without doing a ```docker swarm leave``` first. I'm not sure if the correct order would be to do a ```swarm leave``` before a ```node rm```, but if that's the case, shouldn't:
```node rm``` either result in an error, if the node hasn't left the swarm yet
or shouldn't:
```node rm``` automatically remove the node from the swarm too?

Because as it is right now, it seems that one can get the swarm in a 'broken' state, by doing a ```node rm``` without a ```swarm leave``` first.

I have now fixed it, by re-initializing the whole swarm, and everything seems to be working fine again, but I think that the behavior of ```node rm``` is buggy in the current state and should do one of the above.
